MIGRATING.md

```diff
@@ -0,0 +1,77 @@
+
+## Migrating from 1.x.x to 2.0.0
+
+Version 2.0.0 of this SDK introduces a number of breaking changes from previous versions. 
+In addition to many new classes/interfaces, some existing class/interfaces are no longer backward compatible due to method renaming and signature changes, see [Breaking Changes](#breaking-changes).
+
+### Breaking Changes
+
+Following is a list of changes that break backward compatibility in version 2.0.0.
+
+- `com.okta.sdk.client.Client`
+   - Renamed `createRule` to `createGroupRule`
+   - Renamed `getRule` to `getGroupRule`
+   - Renamed `listRules` to `listGroupRules`
+   
+- `com.okta.sdk.resource.log.LogEventList` 
+   - `getLogs(java.lang.String, java.lang.String, java.lang.String, java.lang.String, java.lang.String)`
+   Signature changed to `getLogs(java.util.Date, java.util.Date, java.lang.String, java.lang.String, java.lang.String)`
+   
+- `com.okta.sdk.resource.group.GroupList` 
+   - `listGroups(java.lang.String,java.lang.String, java.lang.String)`
+   Signature changed to `listGroups(java.lang.String, java.lang.String)`
+
+- `com.okta.sdk.resource.application.JsonWebKey`
+   - `getExpiresAt` Return type changed from `Date` to `String`
+
+- `com.okta.sdk.resource.application.SwaThreeFieldApplicationSettingsApplication`
+   - `getTargetUrl` Renamed to `getTargetURL`
+   - `setTargetUrl` Renamed to `setTargetURL`
+
+- `com.okta.sdk.resource.group.Group`
+   - `getType` Return type changed from `String` to `com.okta.sdk.resource.group.GroupType`
+
+- `com.okta.sdk.resource.group.rule.GroupRule`
+   - Removed `delete` 
+   - Removed `getAllGroupsValid` 
+   - Removed `getEmbedded` 
+   - Removed `setAllGroupsValid`
+
+- `com.okta.sdk.resource.log.LogAuthenticationContext`
+   - `getCredentialProvider` Return type changed from `List` to `com.okta.sdk.resource.log.LogCredentialProvider`
+   - `getCredentialType` Return type changed from `List` to `com.okta.sdk.resource.log.LogCredentialType`
+
+- `com.okta.sdk.resource.policy.Policy`
+   - Removed `createRule`
+
+- `com.okta.sdk.resource.policy.PolicyRule`
+   - Removed `setId`
+
+- `com.okta.sdk.resource.user.factor.VerifyFactorRequest`
+   - Removed `getTokenLifetimeSeconds` & `setTokenLifetimeSeconds`
+
+- `com.okta.sdk.resource.user.Role`
+   - `getType` Return type changed from `String` to `com.okta.sdk.resource.role.RoleType`
+   - `setType` Param type changed from `String` to `com.okta.sdk.resource.role.RoleType`
+
+- `com.okta.sdk.resource.user.User`
+   - Removed `addFactor`
+   - Renamed `addGroupTarget` to `addGroupTargetToRole`
+   - Renamed `addRole` to `assignRole`
+   - Removed `endAllSessions`
+   - `expirePassword` Return type changed from `com.okta.sdk.resource.user.TempPassword` to `com.okta.sdk.resource.user.User`
+   - Removed `forgotPassword`
```

Comment:
What does the user do instead?

Reply:
updated. User must use `resetPassword` to achieve this.